spirv_msl.hpp

```diff
@@ -269,6 +255,44 @@ class CompilerMSL : public CompilerGLSL
 		return capture_output_to_buffer && stage_in_var_id != 0;
 	}
 
+	explicit CompilerMSL(std::vector<uint32_t> spirv);
+	CompilerMSL(const uint32_t *ir, size_t word_count);
+	explicit CompilerMSL(const ParsedIR &ir);
+	explicit CompilerMSL(ParsedIR &&ir);
+
+	// attr is a vertex attribute binding used to match
+	// vertex content locations to MSL attributes. If vertex attributes are provided,
+	// is_msl_vertex_attribute_used() will return true after calling ::compile() if
+	// the location was used by the MSL code.
+	void add_msl_vertex_attribute(const MSLVertexAttr &attr);
```

Comment:
To make the C API wrapping easier, I change this around to be a bit more "set, and query", rather than relying on mutable pointers being set deep in the implementation. @billhollings @cdavis5e, are you okay with this change?

Reply:
I have no objections. Can't speak for @billhollings, though.

Reply:
Yup. Looks fine. I've modified and tested MoltenVK to use the updated "set and query" API wrt vertex attributes and resource bindings...and it all works...and is actually a bit cleaner usage.